File: tests/fixtures/skeleton/py/nested.py
"""Nested functions, closures, async, decorators."""
import asyncio

def outer(a: int) -> int:
    cache: dict = {}

    def inner(b: int) -> int:
        return cache.get(b, a + b)

    return inner(2)

@asyncio.coroutine
def legacy_coro() -> int:
    yield 1

async def fetch_all(urls: list[str]) -> list[str]:
    async def fetch_one(url: str) -> str:
        resp = await http_get(url)
        return resp.body

    results = []
    for u in urls:
        results.append(await fetch_one(u))
    return results

File: tests/fixtures/skeleton/rs/nested.rs
pub fn outer(input: i64) -> i64 {
    let factor = 2i64;
    fn inner(x: i64) -> i64 {
        x * factor
    }
    let closure = |y: i64| {
        let z = inner(y);
        z + 1
    };
    closure(input)
}

pub async fn fetch(url: &str) -> String {
    let body = http_get(url).await;
    body
}
